lib/_stream_readable.js

```diff
@@ -753,12 +758,14 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
   return dest;
 };
 
-function pipeOnDrain(src) {
+function pipeOnDrain(src, dest) {
```

Comment:
You shouldn’t need the extra parameter, the destination is available as `this` inside the event listener.

Reply:
> You shouldn’t need the extra parameter, the destination is available as `this` inside the event listener.

This function will be called directly as `ondrain()`, `this` could be `undefined`, I think adding a dest param may be a better choice?